doc/development/plugins.rst

```diff
@@ -1,397 +1,318 @@
+    class MyDevice(Device):
+        """My Documentation."""
 
-To execute operations on the device, the following methods **must** be defined:
+        def execute(self, circuits: QuantumScriptOrBatch, execution_config: "ExecutionConfig" = DefaultExecutionConfig):
+            return 0.0 if isinstance(circuits, qml.tape.QuantumScript) else tuple(0.0 for c in circuits)
 
-.. currentmodule:: pennylane.QubitDevice
+    dev = MyDevice()
 
-.. autosummary::
+    @qml.qnode(dev)
+    def circuit():
+        return qml.state()
+
+    circuit()
+
+Shots
+-----
+
+While the workflow default for shots is specified by :attr:`~.Device.shots`, the device itself should use
+the number of shots specified in the :attr:`~.QuantumScript.shots` property for each quantum tape.
+By pulling shots dynamically for each circuit, users can efficiently distribute a shot budget across batch of
+circuits.
+
+>>> tape0 = qml.tape.QuantumScript([], [qml.sample(wires=0)], shots=5)
+>>> tape1 = qml.tape.QuantumScript([], [qml.sample(wires=0)], shots=10)
+>>> dev = qml.device('default.qubit')
+>>> dev.execute((tape0, tape1))
+(array([0, 0, 0, 0, 0]), array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0]))
+
+The :class:`~.measurements.Shots` class describes the shots. Users can optionally specify a shot vector, or
+different numbers of shots to use when calculating the final expecation value.
+
+>>> tape0 = qml.tape.QuantumScript([], [qml.expval(qml.PauliX(0))], shots=(5, 500, 1000))
+>>> tape0.shots.shot_vector
+(ShotCopies(5 shots x 1),
+ ShotCopies(500 shots x 1),
+ ShotCopies(1000 shots x 1))
+>>> list(tape0.shots)
+[5, 500, 1000]
+>>> list(tape0.shots.bins())
+[(0, 5), (5, 505), (505, 1505)]
+>>> dev.execute(tape0)
+(0.2, -0.052, -0.014)
+
+The first number ``0.2`` is calculated with 5 shots, the second ``-0.052`` is calculated with 500 shots, and
+``-0.014`` is calculated with 1000 shots.  All 1,505 shots can be requested together in one batch, but the post
+processing into the expecation value is done with shots ``0:5``, ``5:505``, and ``505:1505`` respectively.
+
+Wires
+-----
+
+Devices can now either:
+
+1) Strictly use wires provided by the user on initialization ``device(name, wires=wires)``
+2) Infer the number and ordering of wires provided by the submitted circuit.
+3) Strictly require specific wire labels
+
+Option 2 allows workflows to change the number and labeling of wires over time, but sometimes users want
+to enfore a wire convention and labels. If a user does provide wires, the :meth:`~.devices.Device.preprocess` should
+validate that submitted circuits only have wires in the requested range.
+
+>>> dev = qml.device('default.qubit', wires=1)
+>>> circuit = qml.tape.QuantumScript([qml.CNOT((0,1))], [qml.state()])
+>>> dev.preprocess()[0]((circuit, ))
+WireError: Cannot run circuit(s) of default.qubit as they contain wires not found on the device.
+
+PennyLane wires can be any hashable object, where wire labels are distinguished by their equality and hash.
```

Comment:
I assume this corresponds to option 2 mentioned above? Can we include an example here?

Reply:
wire hashability?  That corresponds to any option.

Reply:
Huh I don't know why it was added to this line but I meant to comment on the `map_to_standard_wires` thing below.